Lib/multiprocessing/synchronize.py

```diff
@@ -270,13 +270,13 @@ def wait(self, timeout=None):
 
     def notify(self, n=1):
         assert self._lock._semlock._is_mine(), 'lock is not owned'
-        assert not self._wait_semaphore.acquire(False)
+        assert not self._wait_semaphore.acquire(False), 'Nobody waiting?'
 
         # to take account of timeouts since last notify*() we subtract
         # woken_count from sleeping_count and rezero woken_count
         while self._woken_count.acquire(False):
             res = self._sleeping_count.acquire(False)
-            assert res
+            assert res, 'Attempting to notify but no sleepers?'
```

Comment:
And here too (it's more of a sophisticated internal detail that's been tested).

Reply:
OK... but what would the assert failing indicate?

Reply:
That the implementation has a bug :-) I don't think we can qualify it any further.

Reply:
Heh. "notify: Bug in sleeping_count.acquire - res should not be False"?

Reply:
Fair enough :-)

Reply:
OK, done.